File: app/src/main/java/hibernate/v2/ringtonerandomizer/ui/receiver/IncomingCallReceiver.java
package hibernate.v2.ringtonerandomizer.ui.receiver;

import android.app.Notification;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.database.sqlite.SQLiteDatabase;
import android.os.Build;
import android.preference.PreferenceManager;
import androidx.core.app.NotificationCompat;
import androidx.core.app.NotificationManagerCompat;
import android.telephony.TelephonyManager;

import hibernate.v2.ringtonerandomizer.C;
import hibernate.v2.ringtonerandomizer.R;
import hibernate.v2.ringtonerandomizer.helper.DBHelper;

public class IncomingCallReceiver extends BroadcastReceiver {

	private Context mContext;
	private DBHelper dbhelper;
	private SQLiteDatabase db;

	@Override
	public void onReceive(Context mContext, Intent intent) {
		C.debug("IncomingCallReceiver onReceive");
		this.mContext = mContext;
		SharedPreferences setting = PreferenceManager.getDefaultSharedPreferences(mContext);
		if (setting.getBoolean("pref_enable", true)) {
			String state = intent.getStringExtra(TelephonyManager.EXTRA_STATE);
			if (TelephonyManager.EXTRA_STATE_RINGING.equals(state)) {
				dbhelper = new DBHelper(mContext);

				String message;
				int result = dbhelper.changeRingtone(mContext, null);
				switch (result) {
					case DBHelper.CHANGE_RINGTONE_RESULT_SUCCESS:
						message = mContext.getString(R.string.changed_ringtone);
						break;
					case DBHelper.CHANGE_RINGTONE_RESULT_COUNT_ZERO:
						message = mContext.getString(R.string.notyet);
						break;
					case DBHelper.CHANGE_RINGTONE_RESULT_COUNT_ONE:
						message = mContext.getString(R.string.changed_ringtone_one);
						break;
					case DBHelper.CHANGE_RINGTONE_RESULT_PERMISSION:
					default:
						message = mContext.getString(R.string.change_ringtone_result_permission);
						break;
				}

				if (setting.getBoolean("pref_changed_notification", true) || Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
					showNotification(message);
				}
				dbhelper.close();
			}
		}
	}

	private void showNotification(String message) {
		Notification notification = new NotificationCompat.Builder(mContext, "Changed Notification")
				.setSmallIcon(R.drawable.shuffle)
				.setContentTitle(mContext.getString(R.string.pref_title_changed_notification))
				.setContentText(message)
				.setPriority(NotificationCompat.PRIORITY_LOW).build();

		NotificationManagerCompat notificationManager = NotificationManagerCompat.from(mContext);
		notificationManager.notify(3, notification);
	}
}

File: app/src/main/java/hibernate/v2/ringtonerandomizer/ui/receiver/IncomingCallReceiver.kt
package hibernate.v2.ringtonerandomizer.ui.receiver

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.telephony.TelephonyManager
import androidx.core.app.NotificationCompat
import androidx.core.app.NotificationManagerCompat
import androidx.preference.PreferenceManager
import hibernate.v2.ringtonerandomizer.R
import hibernate.v2.ringtonerandomizer.helper.DBHelper
import hibernate.v2.ringtonerandomizer.helper.UtilHelper

class IncomingCallReceiver : BroadcastReceiver() {

    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    override fun onReceive(context: Context, intent: Intent) {
        UtilHelper.debug("IncomingCallReceiver onReceive")
        val setting = PreferenceManager.getDefaultSharedPreferences(context)
        if (setting.getBoolean("pref_enable", true)) {
            val state = intent.getStringExtra(TelephonyManager.EXTRA_STATE)
            if (TelephonyManager.EXTRA_STATE_RINGING == state) {
                val dbHelper = DBHelper(context)
                val message: String
                val result = dbHelper.changeRingtone(null)
                message = when (result) {
                    DBHelper.CHANGE_RINGTONE_RESULT_SUCCESS -> context.getString(R.string.changed_ringtone)
                    DBHelper.CHANGE_RINGTONE_RESULT_COUNT_ZERO -> context.getString(R.string.notyet)
                    DBHelper.CHANGE_RINGTONE_RESULT_COUNT_ONE -> context.getString(R.string.changed_ringtone_one)
                    DBHelper.CHANGE_RINGTONE_RESULT_PERMISSION -> context.getString(R.string.change_ringtone_result_permission)
                    else -> context.getString(R.string.change_ringtone_result_permission)
                }
                if (setting.getBoolean("pref_changed_notification", true) || Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    showNotification(context, message)
                }
                dbHelper.close()
            }
        }
    }

    private fun showNotification(context: Context, message: String) {
        val notification = NotificationCompat.Builder(context, "Changed Notification")
                .setSmallIcon(R.drawable.shuffle)
                .setContentTitle(context.getString(R.string.pref_title_changed_notification))
                .setContentText(message)
                .setPriority(NotificationCompat.PRIORITY_LOW).build()
        val notificationManager = NotificationManagerCompat.from(context)
        notificationManager.notify(3, notification)
    }
}